WalletWasabi/Services/Terminate/TerminateService.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Services.Terminate
+{
+	public class TerminateService
+	{
+		private Func<Task> _terminateApplicationAsync;
+
+		private const long TerminateStatusIdle = 0;
+		private const long TerminateStatusInProgress = 1;
+		private const long TerminateFinished = 2;
```

Comment:
```suggestion
		private const long TerminateStatusFinished = 2;
```
?

Reply:
Given that these constants are used only in this file, it may suffice to have them without `Terminate` prefix, that's clear from the class name.